pkg/data/domain/app/service.go

```diff
@@ -118,21 +119,36 @@ func (s *Service) patchVersion(ctx context.Context, namespace string, name strin
 		return microerror.Maskf(invalidTypeError, "unexpected type %T found", a)
 	}
 
+	patch := client.MergeFrom(appCR.DeepCopy())
+
 	// Make sure the requested version is available
 	// Easy way:
-	// (1) Reuse `catalogdata.Get(ctx, options)` to get Catalog with AppCatalogEntry CR using version-specific label selector.
+	// (1) Reuse `catalogdata.GetEntries(ctx, options)` to get Catalog with AppCatalogEntry CR using version-specific label selector.
 	// (2) We only keep AppCatalogEntries CRs for 5 most recent versions. If step (1) returns an error, this does not necessarily
 	//     mean an error. So, change selector to `latest=true` and fetch Catalog CR with AppCatalogEntry CR again. This is to reuse
 	//     the `catalogdata.Get(ctx, options)` again. Catalog CR carries the URL of the given catalog, we can use it as a fallback.
 	// (3) Now, fall back to checking the Helm Repository (Catalog) directly. Use HEAD request for the Chart archive, without fetching
 	//     the whole index.yaml which is more "expensive".
-	err = s.validateVersion(ctx, appCR.Spec.Name, version, appCR.Spec.Catalog, appCR.Spec.CatalogNamespace)
+	err = s.findVersion(ctx, appCR, version, appCR.Spec.Catalog, appCR.Spec.CatalogNamespace)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	patch := client.MergeFrom(appCR.DeepCopy())
-	appCR.Spec.Version = version
+	// Handle Flux reconcile annotation used to suspend reconciliation.
+	accessor, err := meta.Accessor(appCR)
+	if err != nil {
+		return err
+	}
+	annotations := accessor.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+	if suspend {
+		annotations["kustomize.toolkit.fluxcd.io/reconcile"] = "disabled"
+	} else {
+		delete(annotations, "kustomize.toolkit.fluxcd.io/reconcile")
+	}
```

Comment:
We should have the strings `"kustomize.toolkit.fluxcd.io/reconcile"` and `"disabled"` in https://github.com/giantswarm/k8smetadata and reference them.

Reply:
Good point